PEP 700: Additional Fields for the Simple API for Package Indexes (#2840)
python/peps: pep-0700.rst

PEP: 700

> Title: Additional Fields for the Simple API for Package Indexes

Reviewer: Maybe a good idea to make clear in the title that these fields are for the new JSON form of the simple API, e.g. prepending "JSON" or "JSON form" to "Simple API"? Reading the title, I mistakenly assumed it was about the _HTML_ simple API, backporting things from the _new JSON [simple] API_, rather than adding these additional fields (despite seeing the discussion about this some time ago). But up to you of course, as the PEP author :slightly_smiling_face:

Author: Honestly, I originally had that but I removed it because I wanted the title to be reasonably short. It's still a bit too long for my liking...

Reviewer: Ah, I see—good point. It's also got a double "for", which isn't ideal either.

If you want to shorten it, do you like this better? Its a fair bit shorter than the current, adds "JSON" and avoids the double "for":

```suggestion
Title: New Fields for the JSON Simple Package Index API
```

Or what about this? Adding "based" to be clearer/more readible, but still shorter than the current version:

```suggestion
Title: New Fields for the JSON-Based Simple Package Index API
```

Author: Actually, it's a (slightly) open question whether the fields get added to the HTML API as well, so I'm going to say no, let's leave it as it is. There are way more significant questions with the PEP content anyway 😉

Reviewer: Totally fair :slightly_smiling_face: If you do want to shorten here, here are a couple options for you to consider:

```suggestion
Title: Additional Fields for the Simple Package Index API
```

or

```suggestion
Title: New Fields for the Simple Package Index API
```

Anyway, I'll stop nagging you now, sorry!

Author: Paul Moore <p.f.moore@gmail.com>
PEP-Delegate: Donald Stufft <donald@stufft.io>
Discussions-To: https://discuss.python.org/t/pep-700-additional-fields-for-the-simple-api-for-package-indexes/20177
Status: Draft
Type: Standards Track
Topic: Packaging
Content-Type: text/x-rst
Created: 21-Oct-2022
Post-History: `21-Oct-2022 <https://discuss.python.org/t/pep-700-additional-fields-for-the-simple-api-for-package-indexes/20177>`__

Abstract
========

:pep:`691` defined a JSON form for the "Simple Repository API". This allowed
clients to more easily query the data that was previously only available in
HTML, as defined in :pep:`503`.

This proposal adds two *optional* additional fields to the JSON form, which
allow it to be used in place of PyPI's `JSON API <https://warehouse.pypa.io/api-reference/json.html>`__
in a number of situations.

- A field to allow retrieval of a list of all the published versions of a project.
- Fields containing the size and upload time for a project file.

Specification
=============

> The proposal adds three new fields to the JSON form of the "project detail"

Reviewer 2: The abstract says two fields.

```suggestion
The proposal adds two new fields to the JSON form of the "project detail"
```

Author: The abstract's wrong. I fixed one but not the other, and you picked the wrong one to correct 🙂

Fix incoming, thanks for spotting this.
